elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/IndexDsl.scala

```diff
@@ -100,21 +100,24 @@ trait IndexDsl extends DslCommons {
                           numberOfReplicas: Int,
                           analyzerMapping: Analysis,
                           refreshInterval: Int = 1,
-                          preload: Seq[String] = Seq.empty[String])
+                          preload: Seq[String] = Seq.empty[String],
+                          maxResultWindow: Int = 10000)
```

Comment:
lgtm, 
the another approach is to make `maxResultWindow` - `Option[Int]`, if not set the index with the default value (`10k`) is created